src/Network/Haskoin/Store/Data.hs

```diff
@@ -722,6 +808,17 @@ instance BinSerial Transaction where
         put is
         put os
```

Comment:
Convert a `Transaction` into a `TxData` and a `(TxData, IntMap Spender)` using the `fromTransaction` function and serialize that. It will be easier to deserialize using a non-Haskell implementation as it reuses the Bitcoin format for transaction serialization, which is provided by all Bitcoin libraries. Conversely use the `toTransaction` function to deserialize this tuple.